packages/server-side-render/src/server-side-render.js

```diff
@@ -53,12 +53,27 @@ export class ServerSideRender extends Component {
 		if ( null !== this.state.response ) {
 			this.setState( { response: null } );
 		}
-		const { block, attributes = null, urlQueryArgs = {} } = props;
+		const {
+			block,
+			attributes = null,
+			httpMethod = 'GET',
+			urlQueryArgs = {},
+		} = props;
+
+		// If httpMethod is 'POST', send the attributes in the request body instead of the URL.
+		// This allows sending a larger attributes object than in a GET request, where the attributes are in the URL.
+		const isPostRequest = 'POST' === httpMethod;
+		const urlAttributes = isPostRequest ? null : attributes;
+		const path = rendererPath( block, urlAttributes, urlQueryArgs );
+		const data = isPostRequest ? { attributes } : null;
 
-		const path = rendererPath( block, attributes, urlQueryArgs );
 		// Store the latest fetch request so that when we process it, we can
 		// check if it is the current request, to avoid race conditions on slow networks.
-		const fetchRequest = ( this.currentFetchRequest = apiFetch( { path } )
+		const fetchRequest = ( this.currentFetchRequest = apiFetch( {
+			path,
+			data,
+			method: httpMethod,
```

Comment:
I think we should make sure that only POST and GET are used.

```suggestion
			method: isPostRequest ? 'POST' : 'GET',
```

Reply:
Good idea, I'll make that change.

Reply:
Committed in https://github.com/WordPress/gutenberg/pull/21068/commits/6ce918abd8321e986c6bbea2e59325607f2fab40